xbmc/FileItem.h

```diff
@@ -723,6 +723,9 @@ class CFileItemList : public CFileItem
   const std::string &GetContent() const { return m_content; };
 
   void ClearSortState();
+
+  VECFILEITEMS::const_iterator begin() { return m_items.begin(); }
+  VECFILEITEMS::const_iterator end() { return m_items.end(); }
```

Comment:
Something doesn't fit together. You return `const_iterator` but the methods are called `begin()` and `end()` (as opposed to `cbegin()` and `cend()`) and you call `m_items.begin()` and `m_items.end()` which return `iterator`.
So either we want
* `const_iterator` in which case the methods should be `cbegin()` and `cend()` and `const`
* or `iterator` in which case the return type should be fixed

or we want both :-)